src/main/java/edu/wpi/grip/ui/SocketControlView.java

```diff
@@ -0,0 +1,172 @@
+package edu.wpi.grip.ui;
+
+import com.google.common.base.Optional;
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import edu.wpi.grip.core.Socket;
+import edu.wpi.grip.core.SocketHint;
+import edu.wpi.grip.core.events.ConnectionAddedEvent;
+import edu.wpi.grip.core.events.ConnectionRemovedEvent;
+import edu.wpi.grip.core.events.SocketChangedEvent;
+import javafx.beans.property.Property;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.fxml.Initializable;
+import javafx.scene.control.*;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.StackPane;
+import org.controlsfx.control.RangeSlider;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.ResourceBundle;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A JavaFX control that renders a {@link Socket}.  This control shows a label with the socket's identifier, as well as
+ * an optional control such as a slider or a number spinner that lets the user interact with the value in the socket.
+ * <p>
+ * This class both subscribes to and sends {@link SocketChangedEvent}s on the event bus, and acts as an intermediary
+ * between EventBus events and a JavaFX property.
+ */
+public class SocketControlView extends AnchorPane implements Initializable {
+    @FXML
+    private Label identifier;
+
+    @FXML
+    private StackPane controlPane;
+
+    private final EventBus eventBus;
+    private final Socket socket;
+    private final Property valueProperty;
+
+    public SocketControlView(EventBus eventBus, Socket<?> socket) {
+        checkNotNull(eventBus);
+        checkNotNull(socket);
+
+        this.eventBus = eventBus;
+        this.socket = socket;
+        this.valueProperty = new SimpleObjectProperty();
+
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("SocketControl.fxml"));
+            fxmlLoader.setRoot(this);
+            fxmlLoader.setController(this);
+            fxmlLoader.load();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
`RuntimeException`s aren't really supposed to be caught.  I'm throwing a `RuntimeException` here because that should never fail, and if it does, I want to immediately close the program because there's no way it can work.  In practice, this should absolutely only happen if we do something weird during development, so we want to see the program crash and a stacktrace show up.

As for the "much better practice" argument, I'm not sure if I agree.  Since this exception should *theoretically* never be thrown, we don't have a way of naming it (like `FooFailedException` or `BarReturnedErrorException`), since we don't know what the exception means.  Plus I don't like bloating our class hierarchy with a bunch of one-off exceptions.